openedx/core/lib/xblock_utils/__init__.py

```diff
@@ -372,8 +374,23 @@ def add_staff_markup(user, disable_staff_debug_info, block, view, frag, context)
             log.warning("Unable to read field in Staff Debug information", exc_info=True)
             field_contents.append((name, "WARNING: Unable to read field"))
 
+    # Retrieve the installed version of the package the XBlock belongs
+    # to, using inspect and importlib trickery. If this fails for any
+    # reason, don't error out with a 500, but instead show the XBlock
+    # version as None, and log a warning message.
+    block_version = None
+    try:
+        block_module = inspect.getmodule(block)
```

Comment:
So, self-review here. This approach does not reliably work because, apparently (my understanding of this is sketchy, and I haven't found conclusive documentation), `inspect.getmodule()` is not deterministic when it comes to objects that are instances of classes with multiple mixins. It looks to me like it's possible that this can resolve do any module of any class in the object's inheritance chain. In other words, this _might_ return the module we're looking for, or it might not.

And if the detection of the module is unreliable, then getting the package version of that module doesn't produce useful information.

Looks like I need to go back to square one on this one.